grid-client/deployer/node_filter.go

```diff
@@ -20,18 +20,69 @@ import (
 )
 
 // FilterNodes filters nodes using proxy
-func FilterNodes(ctx context.Context, tfPlugin TFPluginClient, options types.NodeFilter, ssdDisks, hddDisks []uint64, rootfs []uint64, optionalLimit ...uint64) ([]types.Node, error) {
-	limit := types.Limit{}
-
-	if len(optionalLimit) > 0 {
-		limit = types.Limit{Size: optionalLimit[0]}
-	}
+func FilterNodes(ctx context.Context, tf TFPluginClient, options types.NodeFilter, ssdDisks, hddDisks []uint64, rootfs []uint64, optionalLimit ...int) ([]types.Node, error) {
+	limit := types.Limit{Randomize: true}
 
 	if options.AvailableFor == nil {
-		twinID := uint64(tfPlugin.TwinID)
+		twinID := uint64(tf.TwinID)
 		options.AvailableFor = &twinID
 	}
 
+	if len(optionalLimit) == 0 {
+		return filterNodes(ctx, tf, options, ssdDisks, hddDisks, rootfs, limit)
+	}
+
+	pagesCount := 2
+	cnt := optionalLimit[0]
+	limit.Size = uint64(cnt)
+
+	if cnt < 100 {
+		pagesCount = 3
+	}
+
+	var nodes []types.Node
+	var allErrors error
+	var wg sync.WaitGroup
+	var lock sync.Mutex
+
+	for i := 0; i < pagesCount; i++ {
```

Comment:
I think we should loop if we didn't reach the count and break if we finished the pages